api/src/main/java/org/apache/cloudstack/api/command/admin/diagnostics/RunDiagnosticsCmd.java

```diff
@@ -116,6 +122,28 @@ public long getEntityOwnerId() {
         return Account.ACCOUNT_ID_SYSTEM;
     }
 
+    @Override
+    public String getEventType() {
+        VirtualMachine vm = _entityMgr.findById(VirtualMachine.class, getId());
+        if (vm.getType() == VirtualMachine.Type.ConsoleProxy) {
```

Comment:
it would be better to use a switch case instead of multiple if else, especially for comparing multiple enums

Reply:
Refactored to switch-case statements